Automattic/Gravatar-SDK-Android
# Split SDK in "core" and "ui" modules

Fixes: #75 

### Description

We want to split the SDK into two different modules:

- `core`: This module contains the main functionality, model classes, etc. It will try to keep external dependencies to a minimum.
- `ui`: In this module, we'll implement all the UI components that developers would be able to use in their apps.

In the UI module, `Detekt` and `Ktlint` have to work as they work in the `core` module.

Dokka will generate documentation for both modules in the same place we were generating it previously:

<img width="837" alt="image" src="https://github.com/Automattic/Gravatar-SDK-Android/assets/6974554/75432cbf-f1f7-45bf-9061-cfaae1174b82">

### Testing Steps

- Compile and execute the demo app
- Review `build.gradle.kts` files and make sure everything makes sense.

### Pending tasks

- We should publish the `ui` module artifact to the A8C repository as we are doing right now with the `core` module. See: https://github.com/Automattic/Gravatar-SDK-Android/pull/41
